core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -1200,6 +1200,21 @@ static RelDataType deriveTypeSplit(SqlOperatorBinding operatorBinding,
           SqlFunctionCategory.STRING)
           .withOperandTypeInference(InferTypes.RETURN_TYPE);
 
+  /** The "CONCAT(arg, ...)" function that concatenates arguments.
+   * For example, "CONCAT('a', 'bc', 'd')" returns "abcd",
+   * "CONCAT(x'61', 'a')" returns "0x6161".
+   *
+   * <p>It accepts at least 1 argument and returns null if any of
+   * the arguments is null, if arguments contain bytestring, would return bytes string. */
```

Comment:
how about "... is null, and returns bytes string if any of the arguments is a bytestring"

Reply:
Make sense, done.